apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/GraphQLCompiler.kt

```diff
@@ -1,52 +1,174 @@
+          filePath = it.filePath,
+          source = QueryDocumentMinifier.minify(it.sourceWithFragments)
+      )
+    }.let {
+      args.operationOutputGenerator.generate(it)
+    }
+
+    check(operationOutput.size == ir.operations.size) {
+      """The number of operation IDs (${operationOutput.size}) should match the number of operations (${ir.operations.size}).
+        |Check that all your IDs are unique.
+      """.trimMargin()
+    }
+
+    if (args.operationOutputFile != null) {
+      args.operationOutputFile.writeText(operationOutput.toJson("  "))
+    }
+
+    val customTypeMap = (introspectionSchema.types.values.filter {
+      it is IntrospectionSchema.Type.Scalar && ScalarType.forName(it.name) == null
+    }.map { it.name } + ScalarType.ID.name)
+        .supportedTypeMap(userCustomTypesMap, generateKotlinModels)
+
+    if (generateKotlinModels) {
       GraphQLKompiler(
           ir = ir,
-          customTypeMap = args.customTypeMap,
-          operationOutput = args.operationOutput,
+          customTypeMap = customTypeMap,
+          operationOutput = operationOutput,
           useSemanticNaming = args.useSemanticNaming,
           generateAsInternal = args.generateAsInternal,
           kotlinMultiPlatformProject = args.kotlinMultiPlatformProject,
           enumAsSealedClassPatternFilters = args.enumAsSealedClassPatternFilters.map { it.toRegex() }
       ).write(args.outputDir)
     } else {
+      val context = CodeGenerationContext(
+          reservedTypeNames = emptyList(),
+          typeDeclarations = ir.typeDeclarations,
+          customTypeMap = customTypeMap,
+          operationOutput = operationOutput,
+          nullableValueType = args.nullableValueType,
+          ir = ir,
+          useSemanticNaming = args.useSemanticNaming,
+          generateModelBuilder = args.generateModelBuilder,
+          useJavaBeansSemanticNaming = args.useJavaBeansSemanticNaming,
+          suppressRawTypesWarning = args.suppressRawTypesWarning,
+          generateVisitorForPolymorphicDatatypes = args.generateVisitorForPolymorphicDatatypes
+      )
+
       ir.writeJavaFiles(
           context = context,
           outputDir = args.outputDir
       )
     }
+
+    args.metadataOutputFile.parentFile.mkdirs()
+    if (args.generateMetadata) {
+      val outgoingMetadata = ApolloMetadata(
+          schema = if (metadata == null) introspectionSchema.wrap() else null,
+          schemaPackageName = schemaPackageName,
+          moduleName = args.moduleName,
+          types = ir.enumsToGenerate + ir.inputObjectsToGenerate,
+          fragments = ir.fragments.filter { ir.fragmentsToGenerate.contains(it.fragmentName) },
+          generateKotlinModels = generateKotlinModels,
+          customTypesMap = args.customTypeMap,
+          pluginVersion = com.apollographql.apollo.compiler.VERSION
+      ).let {
+        if (args.rootProjectDir != null) {
+          it.withRelativeFragments(args.rootProjectDir)
```

Comment:
That's the workaround to not break Gradle build cache and still be able to read back into the file to display nice parse errors.